### PR TITLE
ANDAUTH-75 Show error when empty or wrong URL is submitted in server address

### PR DESCRIPTION
**Pull request by ShridharGoel**
Prevent user from proceeding to the login page with a blank server address. Now error message will also be shown when user enters an invalid URL. Earlier the sign in page was getting opened without validating the URL.

### CODE REVIEW

File: app/src/main/java/org/xwiki/android/sync/activities/SettingServerIpViewFlipper.kt [hunk 1]
```diff
@@ -68,8 +70,8 @@ class SettingServerIpViewFlipper(activity: AuthenticatorActivity, contentRootVie
         val serverAddress = serverEditText.text.toString()
 
         if (TextUtils.isEmpty(serverAddress)) {
-            serverEditText.error = mContext.getString(R.string.error_field_required)
             serverEditText.requestFocus()
+            serverEditText.error = mContext.getString(R.string.error_field_required)
             return null
```
Comment (InsanusMokrassar) [inline app/src/main/java/org/xwiki/android/sync/activities/SettingServerIpViewFlipper.kt]: this return can be placed at the `if`

File: app/src/main/java/org/xwiki/android/sync/activities/SettingServerIpViewFlipper.kt [hunk 2]
```diff
@@ -68,8 +70,8 @@ class SettingServerIpViewFlipper(activity: AuthenticatorActivity, contentRootVie
         val serverAddress = serverEditText.text.toString()
 
         if (TextUtils.isEmpty(serverAddress)) {
-            serverEditText.error = mContext.getString(R.string.error_field_required)
             serverEditText.requestFocus()
+            serverEditText.error = mContext.getString(R.string.error_field_required)
```
Comment (InsanusMokrassar) [inline app/src/main/java/org/xwiki/android/sync/activities/SettingServerIpViewFlipper.kt:67]: Are you sure that it is required change (I mean replace of this line)

**Review by InsanusMokrassar — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/xwiki/android/sync/activities/SettingServerIpViewFlipper.kt (hunk 2 above)
Comment (ShridharGoel) [reply app/src/main/java/org/xwiki/android/sync/activities/SettingServerIpViewFlipper.kt:67]: No, I'll revert it.

File: app/src/main/java/org/xwiki/android/sync/activities/SettingServerIpViewFlipper.kt (hunk 1 above)
Comment (ShridharGoel) [reply app/src/main/java/org/xwiki/android/sync/activities/SettingServerIpViewFlipper.kt]: Sorry, couldn't get you?

Comment (InsanusMokrassar) [reply app/src/main/java/org/xwiki/android/sync/activities/SettingServerIpViewFlipper.kt]:
```
    return if (TextUtils.isEmpty(serverAddress)) {
            serverEditText.requestFocus()
            serverEditText.error = mContext.getString(R.string.error_field_required)
            null
        } else {
            try {
                URL(serverAddress)
                serverAddress
            } catch (e: MalformedURLException) {
                Log.e(SettingServerIpViewFlipper::class.java.simpleName, "Wrong url", e)
                serverEditText.error = mContext.getString(R.string.error_invalid_server)
                serverEditText.requestFocus()
                null
            }

        }
```

**Review by InsanusMokrassar — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/xwiki/android/sync/activities/SettingServerIpViewFlipper.kt (hunk 1 above)
Comment (ShridharGoel) [reply app/src/main/java/org/xwiki/android/sync/activities/SettingServerIpViewFlipper.kt]: Updated.

File: app/src/main/java/org/xwiki/android/sync/activities/SettingServerIpViewFlipper.kt (hunk 2 above)
Comment (ShridharGoel) [reply app/src/main/java/org/xwiki/android/sync/activities/SettingServerIpViewFlipper.kt:67]: Updated.

**Review by InsanusMokrassar — APPROVED** [review]

**Pull request merged**